deeprankcore/trainer.py

```diff
@@ -75,80 +75,110 @@ def __init__( # pylint: disable=too-many-arguments, too-many-branches
 
-            self.target = self.dataset_train.target
-            self.task = self.dataset_train.task
             self.classes = self.dataset_train.classes
-            self.classes_to_idx = self.dataset_train.classes_to_idx
+            self.classes_to_index = self.dataset_train.classes_to_index
             self.optimizer = None
             self.batch_size = batch_size
             self.class_weights = class_weights
             self.shuffle = shuffle
             self.subset = self.dataset_train.subset
-            self.node_features = self.dataset_train.node_features
-            self.edge_features = self.dataset_train.edge_features
-            self.clustering_method = self.dataset_train.clustering_method
             self.epoch_saved_model = None
 
             if self.target is None:
                 raise ValueError("No target set. You need to choose a target (set in GraphDataset) for training.")
 
-            self._check_dataset_equivalence()
             self._load_model()
-
         else:
-            self.pretrained_model = pretrained_model
-            self._load_params()
-
             if self.dataset_train is not None:
-                warnings.warn("Pretrained model loaded: dataset_train will be ignored.")
+                _log.warning("Pretrained model loaded: dataset_train will be ignored.")
             if self.dataset_val is not None:
-                warnings.warn("Pretrained model loaded: dataset_val will be ignored.")
+                _log.warning("Pretrained model loaded: dataset_val will be ignored.")
             if self.neuralnet is None:
                 raise ValueError("No neural network class found. Please add it to complete loading the pretrained model.")
             if self.dataset_test is None:
                 raise ValueError("No dataset_test found. Please add it to evaluate the pretrained model.")
             if self.target is None:
                 raise ValueError("No target set. Make sure the pretrained model explicitly defines the target to train against.")
 
-            self._check_dataset_equivalence()
+            self.pretrained_model_path = pretrained_model
+
+            self._load_params()
             self._load_pretrained_model()
 
+    def _init_output_exporters(self, output_exporters: Optional[List[OutputExporter]]):
+
+        if output_exporters is not None:
+            self._output_exporters = OutputExporterCollection(*output_exporters)
+        else:
+            self._output_exporters = OutputExporterCollection(HDF5OutputExporter('./output'))
+
+    def _init_datasets(self,  # pylint: disable=too-many-arguments
+                       train_dataset: Union[GraphDataset, GridDataset],
+                       validation_dataset: Optional[Union[GraphDataset, GridDataset]],
+                       test_dataset: Optional[Union[GraphDataset, GridDataset]],
```

Comment:
```suggestion
                       dataset_test: Optional[Union[GraphDataset, GridDataset]],
```

Reply:
Changed this in commit 9ad784c964ef77876f71e39599c150dcace6bb22.